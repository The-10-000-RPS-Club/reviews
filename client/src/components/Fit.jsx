import React, { useState } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

function Fit({ rating }) {
  const [score] = useState((rating / 1) * 100);

  return (
    <Wrapper>

      <Title>Overall Fit Rating</Title>
      <Slider>
        <ValueBox inputWidth={score} />
        <Divider />

      </Slider>

      <SmallTag>Runs Small</SmallTag>
      <LargeTag>Runs Large</LargeTag>
    </Wrapper>
  );
}

Fit.propTypes = {
  rating: PropTypes.string.isRequired,
};

const Slider = styled.div`
  position: relative;
  display: inline-block;
  width: 200px;
  height: 8px;
  border: 1px solid #ccc;
  opacity: 0.7;
  margin-bottom: 5px;
`;

const ValueBox = styled.div`
  position: absloute;
  background-color: #2b6692;
  height: 12px;
  width: 12px;
  top: 0;
  left: -6px;
  display: block;
  margin-top: -2px;
  z-index: 2;
  margin-left: ${(props) => props.inputWidth || '0'}%;
  box-shadow: inset 1px 0 0 rgba(255,255,255,.5),inset -1px 0 0 rgba(0,0,0,.25);
  background-image: -webkit-linear-gradient(top,rgba(255,255,255,.3) 0,rgba(255,255,255,0) 100%)
`;
const SmallTag = styled.span`
  font-size: 12px;
  float: left;
`;

const LargeTag = styled.span`
  font-size: 12px;
  float: right;
  text-align: right;
`;
const Wrapper = styled.div`
  font-family: "Roboto","Helvetica Neue","Helvetica","Arial",sans-serif!
`;
const Title = styled.h2`
  display: block!important;
  margin-bottom: .5em!important;
  font-size: 15px;
  line-height: 20px;
  font-style: normal;
  font-weight: 400;
`;
const Divider = styled.span`
  display: block;
  position: absolute;
  top: 0;
  left: 50%;
  width: 2px;
  height: 12px;
  border-right: 1px solid #666;
  margin: -2px 0 0 -1px;
  opacity: 0.7;
  z-index: 0;
`;

export default Fit;
