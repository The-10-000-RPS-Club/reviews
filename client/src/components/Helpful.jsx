import React, { useState, useEffect } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import axios from 'axios';

function Helpful({ helpful, idd, patchData }) {
  const [yes, setYes] = useState(helpful.yes);
  const [no, setNo] = useState(helpful.no);
  const [selected, setSelected] = useState(null);
  const [id] = useState(idd);

  useEffect(() => {
    if (selected !== null) {
      patchData(selected, id);
    }
  }, [selected]);

  return (
    <span>
      Helpful?
      <HelpfulButton onClick={() => {
        setYes(yes + 1);
        setSelected('yes');
      }}
      >
        Yes {yes}
      </HelpfulButton>
      <HelpfulButton onClick={() => {
        setNo(no + 1);
        setSelected('no');
      }}
      >
        No {no}
      </HelpfulButton>
      <HelpfulButton>Report as inappropriate</HelpfulButton>
    </span>
  );
}

Helpful.propTypes = {
  helpful: PropTypes.shape().isRequired,
  idd: PropTypes.string.isRequired,
  patchData: PropTypes.func.isRequired,
};

const HelpfulButton = styled.button`
  background-color: white;
  border: 1px solid grey;
  border-radius: 2px;
  &:hover {
    box-shadow: inset 0 0 3px #000000;
  }
`;

export default Helpful;