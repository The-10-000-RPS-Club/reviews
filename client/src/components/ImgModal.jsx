import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

function ImgModal({ image, showImgModal, onClose }) {
  if (!showImgModal) {
    return null;
  }
  return (
    <Modal>
      <Content>
        <Image inputSrc={image.url} />
        <Description>{image.description}</Description>
      </Content>

        <Button onClick={() => onClose(false)}>X</Button>

    </Modal>
  );
}

ImgModal.propTypes = {
  image: PropTypes.shape().isRequired,
  onClose: PropTypes.func.isRequired,
  showImgModal: PropTypes.bool.isRequired,
};
const Modal = styled.div`
  margin: 0, auto;
  position: absolute;
  top: 25%;
  left: 40%;
  z-index: 10;
  height: auto;
  width: auto;
  background-color: #000;
  box-shadow: 0 10px 25px rgba(0,0,0,.5);
  @supports (offset-rotate: 0deg) {
    offset-rotate: 0deg;
    offset-path: path("M 250,100 S -300,500 -700,-200");
    }
  }
  `;

const Content = styled.div`
  color: white;
  left: 40%;
  padding: 1rem;
`;
const Image = styled.img`
background-attachment: scroll;
width: 500px;
height: 500px;
margin: auto;
position: relative;
top: 50%;
left: 20%;
background-size: cover;
background-repeat: no-repeat;
background-image: url(${(props) => props.inputSrc});
`;
const Description = styled.p`
  text-align: center;
`;
const Button = styled.button`
  position: absolute;
  font-family: "Arial Black", Gadget, sans-serif;
  top: 3px;
  right: 3px;
  border: none;
  font-size: 16px;
  text-align: center;
  color: black;
  background-color: #767676;
  border-radius: 100%;
  height: 30px;
  width: 30px;
  &:hover{
    background-color: transparent;
    background-color: #fff;
  }
`;

export default ImgModal;

// body {
//   background: #eee;
//   display: flex;
//   justify-content: center;
//   align-items: center;
// }

//   .actions {
//     border-top: 1px solid #ccc;
//     background: #eee;
//     padding: 0.5rem 1rem;
//     button {
//       border: 0;
//       background: #78f89f;
//       border-radius: 5px;
//       padding: 0.5rem 1rem;
//       font-size: 0.8rem;
//       line-height: 1;
//     }
//   }
// }

// #centered-toggle-button {
//   position: absolute;
// }
