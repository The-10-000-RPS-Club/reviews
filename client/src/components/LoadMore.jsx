// Will contain the compoent button to load more reviews upond click
// will use a GET request
import React, { useState, useEffect } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

function LoadMore({ changeDisplay, currentDisplay }) {
  const [display, setDisplay] = useState(currentDisplay);

  useEffect(() => {
    changeDisplay(display);
  }, [display]);

  return (
    <Button onClick={() => setDisplay(display + 5)}>Load More</Button>
  );
}

LoadMore.propTypes = {
  currentDisplay: PropTypes.number.isRequired,
  changeDisplay: PropTypes.func.isRequired,
};

const Button = styled.button`
  position: relative!important;
  display: block;
  text-align: center;
  margin: 10px auto;
  width: 33%;
  float: none;
  background-color: white;
  border: 1px solid grey;
  border-radius: 2px;
  padding: 5px 14px;
  font-family: Arial,Helvetica,'Bitstream Vera',sans-serif;
  line-height: 20px;
  font-size: 14px;
  color: #000;
  background-color: #ededed;
  &:hover {
    box-shadow: inset 0 0 1.5px #000000;
  }
`;

export default LoadMore;
