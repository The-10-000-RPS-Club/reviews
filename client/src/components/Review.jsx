/* eslint-disable import/extensions */
import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Star from './Star.jsx';

const FlexContainer = styled.section`
display: flex;
font-family: Arial, Helvetica, sans-serif;
border-bottom: groove;
`;
const FlexBox = styled.div`
  margin: 10px;
  padding: 20px;

`;
const JessesProblem = styled.img`
  width: 150px;
  height: 150px;
  background-image: url(${(props) => props.inputSrc});
`;

const Review = ({ review }) => (
  <FlexContainer>

    <FlexBox>
      <p>{review.nickname}</p>
      <p>{review.location}</p>
    </FlexBox>
    <FlexBox>
      <p>{review.rating}</p>
      <p>{review.title}</p>
      <p>{review.body}</p>
      {review.images[0] ? <JessesProblem inputSrc={review.images[0].url} /> : null}

    </FlexBox>
    <FlexBox>
      <Star rating={review.rating} />
    </FlexBox>

  </FlexContainer>
);

Review.propTypes = {
  review: PropTypes.shape().isRequired,
};

export default Review;

// { <p>
// {review.images[0] ? <img src={review.images[0].url} alt="description" /> : null}
// </p> }