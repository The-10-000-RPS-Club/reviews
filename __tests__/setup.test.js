import React from 'react';
import ReactDOM from 'react-dom';
// import { shallow, mount, render } from 'enzyme';
import { render, fireEvent, getByTestId, waitFor } from '@testing-library/react'
import renderer from 'react-test-renderer';
import ReviewService from '../client/src/components/ReviewService';
import ReviewStatsContainer from '../client/src/components/ReviewStatsContainer';
import ImgModal from '../client/src/components/ImgModal';
import Review from '../client/src/components/Review';
import Helpful from '../client/src/components/Helpful';
import Fit from '../client/src/components/Fit';
import LoadMore from '../client/src/components/LoadMore';
import Star from '../client/src/components/Star';
import SortBy from '../client/src/components/SortBy';
import ReviewModal from '../client/src/components/ReviewModal';
import sampleReview from '../sampleData.js'
const noop = () => { };
Object.defineProperty(window, 'scrollTo', { value: noop, writable: true });

describe('A suite of tests for Reviews Services Container', () => {
  it('renders Reviews Services correctly', () => {
    const tree = renderer
      .create(<ReviewService start={1} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('A suite of tests for ReviewModal Component', () => {
  it('renders Reviews Services correctly', () => {
    const tree = renderer
      .create(<ReviewModal img={sampleReview.images[0]} showReviewModal={true} onClose={() => (null)} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('it does nopt renders Reviews Services', () => {
    const tree = renderer
      .create(<ReviewModal img={sampleReview.images[0]} showReviewModal={false} onClose={() => (null)} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('A suite of tests for Review Display Component', () => {
  it('renders correctly Review Display Component', () => {
    const tree = renderer
      .create(<Review
        review={sampleReview}
        patchData={() => (null)}
        showImgModal={() => (null)}
        setModalImg={() => (null)}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('A suite of tests for Review Stats Container Component', () => {
  it('renders correctly Review Display Component', () => {
    const avg = {
      rating: 4.8,
      fit: .667,
      1: 1,
      2: 3,
      3: 2,
      4: 5,
      5: 4,
      count: 15,
    };
    const tree = renderer
      .create(<ReviewStatsContainer
        avgReview={avg}
        setSort={() => (null)}
        sort={'Most Recent'}
        display={5}
        openReview={() => (null)}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});

describe('A suite of tests for ImgModal Component', () => {
  it('renders correctly ImgModal Component', () => {
    const tree = renderer
      .create(<ImgModal
        image={sampleReview.images[0]}
        showImgModal={true}
        onClose={() => (null)}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('does not renders ImgModal Component', () => {
    const tree = renderer
      .create(<ImgModal
        image={sampleReview.images[0]}
        showImgModal={false}
        onClose={() => (null)}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});


describe('A suite of tests for Load More Component', () => {
  it('renders correctly Load More Button Component', () => {
    const tree = renderer
      .create(<LoadMore
        changeDisplay={() => (null)}
        currentDisplay={1}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('A suite of tests for Star Review Component', () => {
  it('renders correctly Star Review Component with 1 ★\'s', () => {
    const tree = renderer
      .create(<Star rating={1} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly Star Review Component with 3 ★\'s', () => {
    const tree = renderer
      .create(<Star rating={3} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly Star Review Component with 5 ★\'s', () => {
    const tree = renderer
      .create(<Star rating={5} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly Star Review Component with 4.5 ★\'s', () => {
    const tree = renderer
      .create(<Star rating={4.5} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly Star Review Component with no ★\'s', () => {
    const tree = renderer
      .create(<Star rating={0} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});


describe('A suite of tests for SortBy Component', () => {
  it('renders correctly SortBy Review Component by "Highest to Lowest Rating"', () => {
    const tree = renderer
      .create(<SortBy sort={'Highest to Lowest Rating'} setSort={() => (null)} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly SortBy Review Component by "Highest to Lowest Rating"', () => {
    const tree = renderer
      .create(<SortBy sort={'Lowesest to Highest Rating'} setSort={() => (null)} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly SortBy Review Component by "Most Helpful"', () => {
    const tree = renderer
      .create(<SortBy sort={'Most Helpful'} setSort={() => (null)} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly SortBy Review Component by "Most Recent"', () => {
    const tree = renderer
      .create(<SortBy sort={'Most Recent'} setSort={() => (null)} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Dropdown displays proper text when dropdown item is selected", () => {
    let mySort = 'Most Recent';
    const mySetSort = (n) => mySort = n;
    const { container } = render(<SortBy sort={mySort} setSort={mySetSort} />);

    const dropDown = getByTestId(container, "dropDown");
    const content = getByTestId(container, 'content');
    const relevant = getByTestId(container, "relevant");
    const helpful = getByTestId(container, "helpful");
    const highest = getByTestId(container, "high");
    const lowest = getByTestId(container, "low");
    const recent = getByTestId(container, "recent");


    expect(mySort).toMatch('Most Recent');
    fireEvent.click(relevant);
    expect(mySort).toMatch('Most Relevant');
    fireEvent.click(helpful);
    expect(mySort).toMatch('Most Helpful');
    fireEvent.click(highest);
    expect(mySort).toMatch('Highest to Lowest Rating');
    fireEvent.click(lowest);
    expect(mySort).toMatch('Lowest to Highest Rating');
    fireEvent.click(recent);
    expect(mySort).toMatch('Most Recent');

  });

});

describe('A suite of tests for Helpful Component', () => {
  it('renders correctly Helpful Button Component', () => {
    const tree = renderer
      .create(<Helpful
        helpfulyes={sampleReview.helpful_yes}
        helpfulno={sampleReview.helpful_no}
        idd={sampleReview._id}
        patchData={() => (null)}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Increment Yes & No Helpful buttons correctly", () => {
    const { container } = render(<Helpful
      helpfulyes={sampleReview.helpful_yes}
      helpfulno={sampleReview.helpful_no}
      idd={sampleReview._id}
      patchData={() => (null)}
    />);

    const yesClick = getByTestId(container, "yesButton");
    const noClick = getByTestId(container, "noButton");
    expect(yesClick.value).toBe("1");
    expect(noClick.value).toBe("2");
    fireEvent.click(yesClick);
    expect(yesClick.value).toBe("2");
    fireEvent.click(noClick);
    expect(noClick.value).toBe("3");
  });

});


describe('A suite of tests for Fit Component', () => {
  it('renders correctly Fit Display Component as "Runs Small', () => {
    const tree = renderer
      .create(<Fit rating={'0'} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly Fit Display Component as "Runs Large', () => {
    const tree2 = renderer
      .create(<Fit rating={'1'} />)
      .toJSON();
    expect(tree2).toMatchSnapshot();
  });

  it('renders correctly Fit Display Component as "True Fit', () => {
    const tree3 = renderer
      .create(<Fit rating={'0.5'} />)
      .toJSON();
    expect(tree3).toMatchSnapshot();
  });


});