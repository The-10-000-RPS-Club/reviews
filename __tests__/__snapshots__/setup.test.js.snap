// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A suite of tests for Fit Component renders correctly Fit Display Component as "Runs Large 1`] = `
<div
  className="sc-jSFkmK lnDnEi"
>
  <div
    className="sc-gtssRu dWODPw"
  >
    <div
      className="sc-dlnjPT iEAyWe"
    />
    <span
      className="sc-gKAblj dJFrAv"
    />
  </div>
  <span
    className="sc-hKFyIo fNFVfG"
  >
    Runs Small
  </span>
  <span
    className="sc-eCApGN dNfuhr"
  >
    Runs Large
  </span>
</div>
`;

exports[`A suite of tests for Fit Component renders correctly Fit Display Component as "Runs Small 1`] = `
<div
  className="sc-jSFkmK lnDnEi"
>
  <div
    className="sc-gtssRu dWODPw"
  >
    <div
      className="sc-dlnjPT eUUHfX"
    />
    <span
      className="sc-gKAblj dJFrAv"
    />
  </div>
  <span
    className="sc-hKFyIo fNFVfG"
  >
    Runs Small
  </span>
  <span
    className="sc-eCApGN dNfuhr"
  >
    Runs Large
  </span>
</div>
`;

exports[`A suite of tests for Fit Component renders correctly Fit Display Component as "True Fit 1`] = `
<div
  className="sc-jSFkmK lnDnEi"
>
  <div
    className="sc-gtssRu dWODPw"
  >
    <div
      className="sc-dlnjPT dUUXmW"
    />
    <span
      className="sc-gKAblj dJFrAv"
    />
  </div>
  <span
    className="sc-hKFyIo fNFVfG"
  >
    Runs Small
  </span>
  <span
    className="sc-eCApGN dNfuhr"
  >
    Runs Large
  </span>
</div>
`;

exports[`A suite of tests for Helpful Component renders correctly Helpful Button Component 1`] = `
<div>
  Helpful?
  <button
    className="sc-iCoHVE bvCgNS"
    data-testid="yesButton"
    onClick={[Function]}
    value={1}
  >
    Yes ·
     
    1
  </button>
  <button
    className="sc-iCoHVE bvCgNS"
    data-testid="noButton"
    onClick={[Function]}
    value={2}
  >
    No ·
     
    2
  </button>
  <button
    className="sc-iCoHVE bvCgNS"
  >
    Report as inappropriate
  </button>
</div>
`;

exports[`A suite of tests for ImgModal Component does not renders ImgModal Component 1`] = `null`;

exports[`A suite of tests for ImgModal Component renders correctly ImgModal Component 1`] = `
<div
  className="sc-kLokih ibCanN"
>
  <button
    className="sc-gXfWUo hkMISl"
    onClick={[Function]}
  >
    X
  </button>
  <div
    className="sc-iklIKw fulWdc"
  >
    <img
      className="sc-jJMGHv fFYxBt"
    />
    <h3
      className="sc-hiKfjK gEvJoG"
    >
      The Greekest
    </h3>
  </div>
</div>
`;

exports[`A suite of tests for Load More Component renders correctly Load More Button Component 1`] = `
<button
  className="sc-ksluoS djYkNW"
  onClick={[Function]}
>
  Load More
</button>
`;

exports[`A suite of tests for Review Display Component renders correctly Review Display Component 1`] = `
<section
  className="sc-fujyUd bPgFlh"
>
  <div
    className="sc-pNWxx glRlCA"
  >
    <p>
      <b>
        theGreekest
      </b>
    </p>
    <p>
      New York
    </p>
  </div>
  <div
    className="sc-jrsJCI iOXdGf"
  >
    <span
      className="sc-bqGHjH gBkTZR"
    >
      <div
        className="sc-bdnylx ddxNlS"
      />
      3 years ago
    </span>
    <h3
      className="sc-crzoUp ljATUS"
    >
      The Greeks Strike Again
    </h3>
    <p>
      GreekGreekGreekGreekGreekGreekGreekGreekGreekGreekGreekGreekGreekGreekGreekGreekGreek
    </p>
    <p>
      <b>
        Height 
      </b>
      Shorter than 5'
    </p>
    <p>
      <b>
        Weight 
      </b>
      375 - 400 lbs
    </p>
    <p>
      <b>
        Age 
      </b>
      18 to 24
    </p>
    <p>
      <b>
        Primary Usage 
      </b>
      <span
        className="sc-bqGHjH gBkTZR"
      >
        Shirt 
        Pratical 
        #streetWear 
      </span>
    </p>
    <img
      className="sc-iqAbSa jqYKGB"
      onClick={[Function]}
    />
    <div>
      Helpful?
      <button
        className="sc-iCoHVE bvCgNS"
        data-testid="yesButton"
        onClick={[Function]}
        value={1}
      >
        Yes ·
         
        1
      </button>
      <button
        className="sc-iCoHVE bvCgNS"
        data-testid="noButton"
        onClick={[Function]}
        value={2}
      >
        No ·
         
        2
      </button>
      <button
        className="sc-iCoHVE bvCgNS"
      >
        Report as inappropriate
      </button>
    </div>
  </div>
  <div
    className="sc-kEqYlL hWtfSf"
  >
    <h2
      className="sc-dIsAE gEkXsQ"
    >
      Overall Fit Rating
    </h2>
    <div
      className="sc-jSFkmK lnDnEi"
    >
      <div
        className="sc-gtssRu dWODPw"
      >
        <div
          className="sc-dlnjPT dUUXmW"
        />
        <span
          className="sc-gKAblj dJFrAv"
        />
      </div>
      <span
        className="sc-hKFyIo fNFVfG"
      >
        Runs Small
      </span>
      <span
        className="sc-eCApGN dNfuhr"
      >
        Runs Large
      </span>
    </div>
  </div>
</section>
`;

exports[`A suite of tests for Review Stats Container Component renders correctly Review Display Component 1`] = `
<div
  className="sc-giAqnE igbTra"
>
  <h1
    className="sc-ezzayL gJDqIQ"
  >
    Reviews
  </h1>
  <button
    className="sc-bYwzba cCGNpU"
    onClick={[Function]}
  >
    Write a review
  </button>
  <section
    className="sc-bCwgka booarE"
  >
    <div
      className="sc-dlMBXb gabvLs"
    >
      <table
        className="sc-iwaifL kywOwR"
      >
        <thead>
          <tr>
            <td>
              Rating Snapshot
            </td>
          </tr>
          <tr>
            <td>
              Select a row below to filter reviews.
            </td>
          </tr>
          <tr>
            <td
              className="sc-iJCRLp cBIuiF"
            >
              5 ★
              <div
                className="sc-kfYqjs igEuCT"
                onClick={[Function]}
              >
                <div
                  className="sc-fKgIGh eeeCjv"
                />
              </div>
              4
            </td>
          </tr>
          <tr>
            <td
              className="sc-iJCRLp cBIuiF"
            >
              4 ★
              <div
                className="sc-kfYqjs igEuCT"
                onClick={[Function]}
              >
                <div
                  className="sc-fKgIGh fQxxPX"
                />
              </div>
              5
            </td>
          </tr>
          <tr>
            <td
              className="sc-iJCRLp cBIuiF"
            >
              3 ★
              <div
                className="sc-kfYqjs igEuCT"
                onClick={[Function]}
              >
                <div
                  className="sc-fKgIGh kELwhx"
                />
              </div>
              2
            </td>
          </tr>
          <tr>
            <td
              className="sc-iJCRLp cBIuiF"
            >
              2 ★
              <div
                className="sc-kfYqjs igEuCT"
                onClick={[Function]}
              >
                <div
                  className="sc-fKgIGh exikfv"
                />
              </div>
              3
            </td>
          </tr>
          <tr>
            <td
              className="sc-iJCRLp cBIuiF"
            >
              1 ★
              <div
                className="sc-kfYqjs igEuCT"
                onClick={[Function]}
              >
                <div
                  className="sc-fKgIGh jsbgTq"
                />
              </div>
              1
            </td>
          </tr>
        </thead>
      </table>
    </div>
    <div
      className="sc-hHEjAm bXHEUy"
    >
      <table
        className="sc-iwaifL kywOwR"
      >
        <caption>
          Average Customer Ratings
        </caption>
        <thead>
          <tr>
            <td>
              Overall
            </td>
            <td
              className="sc-cxNIbT lbEgEG"
            >
              <div
                className="sc-bdnylx dHfqiF"
              />
            </td>
            <td
              className="sc-cxNIbT lbEgEG"
            >
              4.8
            </td>
          </tr>
          <tr>
            <td
              className="sc-lmgQde cPrNCv"
            >
              Overall Fit Rating
            </td>
            <td
              className="sc-cxNIbT lbEgEG"
            >
              <div
                className="sc-jSFkmK lnDnEi"
              >
                <div
                  className="sc-gtssRu dWODPw"
                >
                  <div
                    className="sc-dlnjPT dfYhMK"
                  />
                  <span
                    className="sc-gKAblj dJFrAv"
                  />
                </div>
                <span
                  className="sc-hKFyIo fNFVfG"
                >
                  Runs Small
                </span>
                <span
                  className="sc-eCApGN dNfuhr"
                >
                  Runs Large
                </span>
              </div>
            </td>
          </tr>
        </thead>
      </table>
    </div>
    <div
      className="sc-dIvqjp fEBpLv"
    >
      <p
        className="sc-bkbjAj kHLutn"
      >
        1–
        5
         
        of
         
        15
         
        Reviews
      </p>
    </div>
    <div
      className="sc-iemXMA iPyMCC"
    >
      <div
        className="sc-fnVYJo iUPTYC"
        data-testid="dropDown"
      >
        Sort By:
         
        Most Recent
         
        ▼
        <div
          className="sc-hBMVcZ cGajHf"
          data-testid="content"
        >
          <div
            className="sc-fFSRdu fmPRGV"
            data-testid="relevant"
            onClick={[Function]}
          >
            Most Relevant
          </div>
          <div
            className="sc-fFSRdu fmPRGV"
            data-testid="helpful"
            onClick={[Function]}
          >
            Most Helpful
          </div>
          <div
            className="sc-fFSRdu fmPRGV"
            data-testid="high"
            onClick={[Function]}
          >
            Highest to Lowest Rating
          </div>
          <div
            className="sc-fFSRdu fmPRGV"
            data-testid="low"
            onClick={[Function]}
          >
            Lowest to Highest Rating
          </div>
          <div
            className="sc-fFSRdu fmPRGV"
            data-testid="recent"
            onClick={[Function]}
          >
            Most Recent
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`A suite of tests for ReviewModal Component it does nopt renders Reviews Services 1`] = `null`;

exports[`A suite of tests for ReviewModal Component renders Reviews Services correctly 1`] = `
<div
  className="sc-cBoprd dYZyTo"
>
  <button
    className="sc-iTVIwl aTpSw"
    onClick={[Function]}
  >
    X
  </button>
  <div
    className="sc-jcwofb gtUsti"
  >
    <img
      className="sc-carGAA iNEvDa"
    />
    The Greekest
  </div>
  <div
    className="sc-ciSmjq RpyRj"
  >
    My Review for REI Co-op Men's Sahara Plaid Long-Sleeve Shirt
  </div>
  <div
    className="sc-ciSmjq RpyRj"
  >
    Product rating*
  </div>
  <div
    className="sc-ciSmjq RpyRj"
  >
    Review title*
  </div>
  <div
    className="sc-ciSmjq RpyRj"
  >
    Review*
  </div>
  <div
    className="sc-ciSmjq RpyRj"
  >
    Would you recommend this product to a friend?
  </div>
  <div
    className="sc-ciSmjq RpyRj"
  >
    Overall Fit
  </div>
  <div
    className="sc-ciSmjq RpyRj"
  >
    Best for:
  </div>
  <div
    className="sc-ciSmjq RpyRj"
  >
    Height
  </div>
  <div
    className="sc-ciSmjq RpyRj"
  >
    Weight Range
  </div>
  <div
    className="sc-ciSmjq RpyRj"
  >
    Age
  </div>
  <div
    className="sc-ciSmjq RpyRj"
  >
    Nickname* Location
  </div>
  <div
    className="sc-ciSmjq RpyRj"
  >
    Email*
  </div>
  <div
    className="sc-ciSmjq RpyRj"
  >
    <button
      className="sc-iBzFoy ebpcog"
    >
      Post Review
    </button>
  </div>
</div>
`;

exports[`A suite of tests for Reviews Services Container renders Reviews Services correctly 1`] = `
<p>
  Be the first to review this product
</p>
`;

exports[`A suite of tests for SortBy Component renders correctly SortBy Review Component by "Highest to Lowest Rating" 1`] = `
<div
  className="sc-fnVYJo iUPTYC"
  data-testid="dropDown"
>
  Sort By:
   
  Highest to Lowest Rating
   
  ▼
  <div
    className="sc-hBMVcZ cGajHf"
    data-testid="content"
  >
    <div
      className="sc-fFSRdu fmPRGV"
      data-testid="relevant"
      onClick={[Function]}
    >
      Most Relevant
    </div>
    <div
      className="sc-fFSRdu fmPRGV"
      data-testid="helpful"
      onClick={[Function]}
    >
      Most Helpful
    </div>
    <div
      className="sc-fFSRdu fmPRGV"
      data-testid="high"
      onClick={[Function]}
    >
      Highest to Lowest Rating
    </div>
    <div
      className="sc-fFSRdu fmPRGV"
      data-testid="low"
      onClick={[Function]}
    >
      Lowest to Highest Rating
    </div>
    <div
      className="sc-fFSRdu fmPRGV"
      data-testid="recent"
      onClick={[Function]}
    >
      Most Recent
    </div>
  </div>
</div>
`;

exports[`A suite of tests for SortBy Component renders correctly SortBy Review Component by "Highest to Lowest Rating" 2`] = `
<div
  className="sc-fnVYJo iUPTYC"
  data-testid="dropDown"
>
  Sort By:
   
  Lowesest to Highest Rating
   
  ▼
  <div
    className="sc-hBMVcZ cGajHf"
    data-testid="content"
  >
    <div
      className="sc-fFSRdu fmPRGV"
      data-testid="relevant"
      onClick={[Function]}
    >
      Most Relevant
    </div>
    <div
      className="sc-fFSRdu fmPRGV"
      data-testid="helpful"
      onClick={[Function]}
    >
      Most Helpful
    </div>
    <div
      className="sc-fFSRdu fmPRGV"
      data-testid="high"
      onClick={[Function]}
    >
      Highest to Lowest Rating
    </div>
    <div
      className="sc-fFSRdu fmPRGV"
      data-testid="low"
      onClick={[Function]}
    >
      Lowest to Highest Rating
    </div>
    <div
      className="sc-fFSRdu fmPRGV"
      data-testid="recent"
      onClick={[Function]}
    >
      Most Recent
    </div>
  </div>
</div>
`;

exports[`A suite of tests for SortBy Component renders correctly SortBy Review Component by "Most Helpful" 1`] = `
<div
  className="sc-fnVYJo iUPTYC"
  data-testid="dropDown"
>
  Sort By:
   
  Most Helpful
   
  ▼
  <div
    className="sc-hBMVcZ cGajHf"
    data-testid="content"
  >
    <div
      className="sc-fFSRdu fmPRGV"
      data-testid="relevant"
      onClick={[Function]}
    >
      Most Relevant
    </div>
    <div
      className="sc-fFSRdu fmPRGV"
      data-testid="helpful"
      onClick={[Function]}
    >
      Most Helpful
    </div>
    <div
      className="sc-fFSRdu fmPRGV"
      data-testid="high"
      onClick={[Function]}
    >
      Highest to Lowest Rating
    </div>
    <div
      className="sc-fFSRdu fmPRGV"
      data-testid="low"
      onClick={[Function]}
    >
      Lowest to Highest Rating
    </div>
    <div
      className="sc-fFSRdu fmPRGV"
      data-testid="recent"
      onClick={[Function]}
    >
      Most Recent
    </div>
  </div>
</div>
`;

exports[`A suite of tests for SortBy Component renders correctly SortBy Review Component by "Most Recent" 1`] = `
<div
  className="sc-fnVYJo iUPTYC"
  data-testid="dropDown"
>
  Sort By:
   
  Most Recent
   
  ▼
  <div
    className="sc-hBMVcZ cGajHf"
    data-testid="content"
  >
    <div
      className="sc-fFSRdu fmPRGV"
      data-testid="relevant"
      onClick={[Function]}
    >
      Most Relevant
    </div>
    <div
      className="sc-fFSRdu fmPRGV"
      data-testid="helpful"
      onClick={[Function]}
    >
      Most Helpful
    </div>
    <div
      className="sc-fFSRdu fmPRGV"
      data-testid="high"
      onClick={[Function]}
    >
      Highest to Lowest Rating
    </div>
    <div
      className="sc-fFSRdu fmPRGV"
      data-testid="low"
      onClick={[Function]}
    >
      Lowest to Highest Rating
    </div>
    <div
      className="sc-fFSRdu fmPRGV"
      data-testid="recent"
      onClick={[Function]}
    >
      Most Recent
    </div>
  </div>
</div>
`;

exports[`A suite of tests for Star Review Component renders correctly Star Review Component with 1 ★'s 1`] = `
<div
  className="sc-bdnylx blTEpM"
/>
`;

exports[`A suite of tests for Star Review Component renders correctly Star Review Component with 3 ★'s 1`] = `
<div
  className="sc-bdnylx cSglpA"
/>
`;

exports[`A suite of tests for Star Review Component renders correctly Star Review Component with 4.5 ★'s 1`] = `
<div
  className="sc-bdnylx dDgCkX"
/>
`;

exports[`A suite of tests for Star Review Component renders correctly Star Review Component with 5 ★'s 1`] = `
<div
  className="sc-bdnylx gulRub"
/>
`;

exports[`A suite of tests for Star Review Component renders correctly Star Review Component with no ★'s 1`] = `
<div
  className="sc-bdnylx eTIvCG"
/>
`;
